packages/tools/src/tools/segmentation/strategies/utils/getStrategyData.ts

```diff
@@ -0,0 +1,65 @@
+import { cache } from '@cornerstonejs/core';
+import { isVolumeSegmentation } from './stackVolumeCheck';
+import { LabelmapToolOperationDataStack } from '../../../../types';
+
+function getStrategyData({ operationData, viewport }) {
+  let segmentationImageData, segmentationScalarData, imageScalarData;
+
+  if (isVolumeSegmentation(operationData)) {
```

Comment:
Could you drop both halfs of the strategy into individual functions?  It would make the overall function less complex, and that makes it easier to understand.